cmd/telegraf/telegraf.go

```diff
@@ -271,7 +271,7 @@ func reloadLoop(stop chan struct{}, s service.Service) {
 		log.Printf("I! Tags enabled: %s", c.ListTags())
 
 		if *fPidfile != "" {
-			f, err := os.Create(*fPidfile)
+			f, err := os.OpenFile(*fPidfile, os.O_EXCL|os.O_CREATE|os.O_WRONLY, 0666)
```

Comment:
> So it gets messy really fast.

Agreed. :) I thought about this when I started, and out of ~~laziness~~complexity avoidance, I went the easier, "operator intervention required" route.

I'm sure there are other apps that won't boot if there's lingering lock after a crash (`apt`), but I don't know if that's a good pattern to follow.